src/main/java/org/openrewrite/gradle/RecipeDependenciesExtension.java

```diff
@@ -42,11 +42,46 @@ public RecipeDependenciesExtension(ConfigurationContainer configurationContainer
 
     @SuppressWarnings("unused")
     public void parserClasspath(String dependencyNotation) {
-        dependencies.add(dependencyNotation);
+        addDependencyForSourceSet("main", dependencyNotation);
+    }
+
+    @SuppressWarnings("unused")
+    public void testParserClasspath(String dependencyNotation) {
+        addDependencyForSourceSet("test", dependencyNotation);
+    }
+
+    @SuppressWarnings("unused")
+    public void integrationTestParserClasspath(String dependencyNotation) {
+        addDependencyForSourceSet("integrationTest", dependencyNotation);
+    }
+
```

Comment:
I don't think we use this pattern, so then let's limit what we add here for now.
```suggestion
```

Reply:
Removed. This method (as well as the other two are unused anyway, as this gets called dynamically).